hawkular-inventory-rest-api/src/main/java/org/hawkular/inventory/rest/RestBulk.java

```
@@ -0,0 +1,510 @@
+
+        return Response.status(CREATED).entity(statuses).build();
+    }
+
+    private Map<ElementType, Map<CanonicalPath, Integer>> bulkCreate(
+            Map<String, Map<ElementType, List<Object>>> entities, CanonicalPath rootPath) {
+
+        Map<ElementType, Map<CanonicalPath, Integer>> statuses = new HashMap<>();
+
+        TransactionFrame transaction = inventory.newTransactionFrame();
+        Inventory binv = transaction.boundInventory();
+
+        IdExtractor idExtractor = new IdExtractor();
+
+        try {
+            for (Map.Entry<String, Map<ElementType, List<Object>>> e : entities.entrySet()) {
+                Map<ElementType, List<Object>> allBlueprints = e.getValue();
+
+                CanonicalPath parentPath = canonicalize(e.getKey(), rootPath);
+
+                ResolvableToSingle<?, ?> single = binv.inspect(parentPath, ResolvableToSingle.class);
+
+                for (Map.Entry<ElementType, List<Object>> ee : allBlueprints.entrySet()) {
+                    ElementType elementType = ee.getKey();
+                    List<Object> rawBlueprints = ee.getValue();
+
+                    List<Blueprint> blueprints = deserializeBlueprints(elementType, rawBlueprints);
+
+                    if (elementType == ElementType.relationship) {
+                        bulkCreateRelationships(statuses, parentPath,
+                                (ResolvableToSingleWithRelationships<?, ?>) single, elementType, blueprints);
+                    } else {
+                        bulkCreateEntity(statuses, idExtractor, parentPath, single, elementType, blueprints);
+                    }
+                }
+            }
+            transaction.commit();
+            return statuses;
+        } catch (Throwable t) {
+            // TODO this potentially leaves behind the security resources of the entities that have been created
+            // before the transaction failure
```

it looks like `bulkCreateEntity()` swallows all the exceptions, perhaps `bulkCreateRelationships()` could have the same Exception -> http 500 thing.

One thinkable way of reaching this place is the situation when the path is wrong (line 355). Then the inner for-loop doesn't make sense.. Perhaps, setting all the status codes new entities on this non-existent path to 404 could work.

Actually, I think this method should be as robust as possible and the rollback is probably an overkill. Btw. is it possible to create an entity and in the same bulk operation also another entity under it (parent-child resources for instance)?